classes/kohana/encrypt.php

```diff
@@ -1,213 +1,134 @@
 <?php defined('SYSPATH') or die('No direct script access.');
 /**
  * The Encrypt library provides two-way encryption of text and binary strings
- * using the [Mcrypt](http://php.net/mcrypt) extension, which consists of three
- * parts: the key, the cipher, and the mode.
- *
- * The Key
- * :  A secret passphrase that is used for encoding and decoding
- *
- * The Cipher
- * :  A [cipher](http://php.net/mcrypt.ciphers) determines how the encryption
- *    is mathematically calculated. By default, the "rijndael-128" cipher
- *    is used. This is commonly known as "AES-128" and is an industry standard.
- *
- * The Mode
- * :  The [mode](http://php.net/mcrypt.constants) determines how the encrypted
- *    data is written in binary form. By default, the "nofb" mode is used,
- *    which produces short output with high entropy.
- *
+ * using the [Sodium](https://www.php.net/manual/en/book.sodium.php) extension.
+
  * @package    Kohana
  * @category   Security
  * @author     Kohana Team
  * @copyright  (c) 2007-2012 Kohana Team
  * @license    http://kohanaframework.org/license
  */
 class Kohana_Encrypt {
-
 	/**
-	 * @var  string  default instance name
+	 * @var string
 	 */
 	public static $default = 'default';
 
 	/**
-	 * @var  array  Encrypt class instances
+	 * @var array
 	 */
-	public static $instances = array();
+	public static $instances = [];
 
 	/**
-	 * @var  string  OS-dependent RAND type to use
+	 * @var string
 	 */
-	protected static $_rand;
+	private $key;
 
 	/**
 	 * Returns a singleton instance of Encrypt. An encryption key must be
 	 * provided in your "encrypt" configuration file.
 	 *
-	 *     $encrypt = Encrypt::instance();
+	 * @param string $name configuration group name
 	 *
-	 * @param   string  $name   configuration group name
-	 * @return  Encrypt
+	 * @return self
 	 */
-	public static function instance($name = NULL)
+	public static function instance($name = null)
 	{
-		if ($name === NULL)
+		if ($name === null)
 		{
-			// Use the default instance name
-			$name = Encrypt::$default;
+			$name = self::$default;
 		}
 
-		if ( ! isset(Encrypt::$instances[$name]))
+		if (!isset(self::$instances[$name]))
 		{
 			// Load the configuration data
-			$config = Kohana::$config->load('encrypt')->$name;
+			$config = \Kohana::$config->load('encrypt')->$name;
```

Comment:
```suggestion
			$config = Kohana::$config->load('encrypt')->$name;
```

Not needed as the code is not in a namespace.